File: DynamicKey/AgoraDynamicKey/python/sample/sample.py
#! /usr/bin/python
# ! -*- coding: utf-8 -*-

import argparse
import sys
import os
import time
from random import randint

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
from DynamicKey4 import generateRecordingKey
from DynamicKey4 import generateMediaChannelKey


def main():
    unixts = int(time.time())
    randomint = -2147483647

    parser = argparse.ArgumentParser(description='Generator for channel key')
    parser.add_argument("--app_id", dest="app_id", default="")
    parser.add_argument("--certificate", dest="certificate", default="")
    parser.add_argument("--channel_name", dest="channel_name", default="")
    parser.add_argument("--uid", dest="uid", type=int, default=0)
    parser.add_argument("--expired_ts", dest="expired_ts", type=int, default=0)

    args = parser.parse_args()

    if not args.app_id:
        print >> sys.stderr, "No APP ID is designated"
        return -1

    if not args.certificate:
        print >> sys.stderr, "No certificate is designated"
        return -1

    if not args.channel_name:
        print >> sys.stderr, "No channel name is degsignated"
        return -1

    print "Recording key:", generateRecordingKey(args.app_id, args.certificate, \
                                                 args.channel_name, unixts, randomint, args.uid, args.expired_ts)

    print "Channel key:", generateMediaChannelKey(args.app_id, args.certificate, \
                                                  args.channel_name, unixts, randomint, args.uid, args.expired_ts)


if __name__ == "__main__":
    main()

File: DynamicKey/AgoraDynamicKey/nodejs/sample/sample.js
var DynamicKey4 = require('../src/DynamicKey4');
var appID  = "970ca35de60c44645bbae8a215061b33";
var appCertificate     = "5cfd2fd1755d40ecb72977518be15d3b";
var channel = "my channel name";
var ts = Math.floor(new Date() / 1000);
var r = Math.floor(Math.random() * 0xFFFFFFFF);
var uid = 2882341273;
var expiredTs = 0;

console.log("4 recording key: " + DynamicKey4.generateRecordingKey(appID, appCertificate, channel, ts, r, uid, expiredTs));
console.log("4 channel key: " + DynamicKey4.generateMediaChannelKey(appID, appCertificate, channel, ts, r, uid, expiredTs));
